From a12da682282322bfa8c2a4307bb45921f346f37c Mon Sep 17 00:00:00 2001
From: Aloka Dixit <quic_alokad@quicinc.com>
Date: Tue, 6 Jul 2021 11:13:15 -0700
Subject: [PATCH 08/27] nl80211: ensure HE capabilities are set for EHT

EHT capabilities can only be set if HE capabilities are set as well.
Return error otherwise.

Signed-off-by: Aloka Dixit <quic_alokad@quicinc.com>
---
 net/wireless/nl80211.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -7308,6 +7308,10 @@ static int nl80211_new_station(struct sk
 	    (params.link_sta_params.ht_capa || params.link_sta_params.vht_capa))
 		return -EINVAL;
 
+	/* Ensure that HE capabilities are set along with EHT */
+	if (params.link_sta_params.eht_capa && !params.link_sta_params.he_capa)
+		return -EINVAL;
+
 	/* When you run into this, adjust the code below for the new flag */
 	BUILD_BUG_ON(NL80211_STA_FLAG_MAX != 7);
 
